authentication/rancher/authentication.go

```diff
@@ -150,3 +156,36 @@ func (p *Provider) authDisabled() bool {
 
 	return setting.Value == "false"
 }
+
+func (p *Provider) isAdmin(token string) bool {
+	req, err := http.NewRequest("GET", p.url+"/accounts", nil)
```

Comment:
Actually yeah, returning errors would be better. I did it this way when I thought accessing `/accounts` would give a 401.